File: Easy/Python/1413. Minimum Value to Get Positive Step by Step Sum.py
class Solution(object):
    def minStartValue(self, nums):
        this_sum = 0

        for i in range(len(nums)):
            this_sum += nums[i]
            nums[i] = this_sum
        
        nums.sort()

        if 0 < nums[0]:
            return 1
        else:
            return abs(nums[0]) + 1

File: Easy/Java/1413. Minimum Value to Get Positive Step by Step Sum.java
class Solution {
    public int minStartValue(int[] nums) {
        int sum = 0;

        for(int i = 0; i < nums.length; i++){
            sum += nums[i];
            nums[i] = sum;
        }

        Arrays.sort(nums);
        
        if(0 < nums[0])
            return 1;
        else
            return Math.abs(nums[0]) + 1;
    }
}
